### PR TITLE
[CI] Deploy qtime backend to production on AWS Lightsail

### PR DESCRIPTION
**Pull request by benjaminkostiuk**
## Overview

Adds steps to the CI pipeline to deploy the backend to AWS Lightsail on push to master. Send a message to discord once the new version is running at https://queuetime.tech.

This change is a

- [ ] Bug fix
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that will change exisiting functionality)
- [ ] Documentation update

## Description

 - Add a new deploy action that runs typechecking and tests before deploying to AWS.
 - Uses docker to build an image to deploy so new Dockerfile and .dockerignore
 - Updated `manage.py` to add production options with `ENV` variable.
 - Updated README with instructions.

## Motivation/Links
Automated deployments of the backend allows testing of preview builds with apis enabled.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - master
```
Comment (ruler225) [inline .github/workflows/deploy.yml:6]: Our mainline branch is main, not master

**Review by ruler225 — CHANGES_REQUESTED** [review]
Awesome stuff! Just one minor thing, we don't use a master branch, our mainline branch is main (unless this is intentional and you intend to create a separate master branch for the live deployment?)

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (benjaminkostiuk) [reply .github/workflows/deploy.yml:6]: This is Github Co-pilot giving me the wrong hints and me forgetting we use master lol

File: README.md [hunk 2]
```diff
@@ -128,11 +130,62 @@ Example commands:
...
+
 This will add a new boolean field to all documents in the users collection called `has_completed_onboarding` with a default value of true.
 
-Command line help for the utility is also available by providing the `-h` or `--help` argument.
\ No newline at end of file
+Command line help for the utility is also available by providing the `-h` or `--help` argument.
+
+## Deploying
+
+The QTime backend is deployed to [AWS Lightsail](https://aws.amazon.com/lightsail/) automatically on push to master using the `Deploy` workflow. The steps to manually deploy are below:
```
Comment (ruler225) [inline README.md:146]: Don't forget to update the README too

**Review by ruler225 — APPROVED** [review]
Awesome possum!

**Review by dizona — APPROVED** [review]

**Pull request merged**